### ISSUE TITLE
口红？

### ISSUE THREAD

**Issue by zhaipro** [zhaipro/unet#4]
为什么又开始玩口红了？

**Comment by zhaipro** [comment]
![s](https://user-images.githubusercontent.com/8620842/79929442-4d893100-8478-11ea-9289-81f6ade59e60.jpg)

**Comment by zhaipro** [comment]
重温游戏引擎中提到的光照

**Comment by zhaipro** [comment]
1. http://www.globalblue.cn/destinations/italy/milan/18-shades-of-red-lips/#slide1
1. https://ai.meitu.com/doc?id=9
1. https://ai.meitu.com/algorithm/imageRecognition/faceSegment
1. https://help.aliyun.com/document_detail/155014.html
1. http://mmlab.ie.cuhk.edu.hk/projects/CelebA.html
1. [BiSeNet](https://arxiv.org/pdf/1808.00897.pdf)
1. https://zhuanlan.zhihu.com/p/41475332
1. https://zhuanlan.zhihu.com/p/41679153
1. https://keras.io/zh/
1. https://github.com/zllrunning/face-parsing.PyTorch
1. https://github.com/kirilcvetkov92/Semantic-Segmentation-BiSeNet
1. https://github.com/ooooverflow/BiSeNet
1. https://github.com/petrosgk/Kaggle-Carvana-Image-Masking-Challenge
1. https://towardsdatascience.com/epoch-vs-iterations-vs-batch-size-4dfb9c7ce9c9
1. https://github.com/keras-team/keras-contrib
1. [Xception](https://arxiv.org/pdf/1610.02357.pdf)

**Comment by zhaipro** [comment]
|版本|loss|dice_coeff|psnr|val_loss|val_dice_coeff|val_psnr|
|--|--|--|--|--|--|--|
|2.0.0|0.0047|0.8735|29.3643|0.0046|0.7376|28.0056|
|2.0.1|0.0045|0.8829|29.6047|0.0042|0.6776|27.3260|
|2.0.2|0.0868|0.9132|28.2183|0.1161|0.5191|24.912|
|2.0.3|0.0894|0.9184|28.9648|0.0572|0.7362|27.1730|
|2.0.4|0.0868|0.9203|30.5728|0.1535|0.5305|24.6504|
|2.0.5|0.1170|0.8935|28.7663|0.0746|0.4990|25.1946|
|2.0.6|0.1085|0.8997|31.2613|0.0854|0.6298|26.0387|
|2.0.7|0.0942|0.9128|31.9750|0.0581|0.9050|29.6066|
|2.0.7.1|0.1038|0.9051|31.2448|0.0573|0.9143|29.9382|
|2.0.7.3|0.1027|0.9063|31.3688|0.0578|0.9173|29.9247|
|2.0.7.4|0.1044|0.9049|31.2056|0.0257|0.9250|29.5189|
|2.0.7.5|0.0930|0.9180|52.0262|0.0662|0.9085|29.2761|
|2.0.8|0.0929|0.9139|32.0576|0.0607|0.8951|29.5029|
|2.0.8.1|0.0990|0.9118||0.0443|0.9270|
|2.0.8.2|0.0901|0.9201||0.0747|0.9252|
|2.0.8.3|0.0928|0.9176||0.0714|0.9275|
|2.0.8.4|0.0917|0.9220||0.0702|0.9275|
|2.0.8.5|0.0879|0.9252||0.0696|0.9303|
|2.0.9|0.0929|0.9140|32.0216|0.0581|0.8966|29.6190|
|2.1.1|0.1201||0.8895|30.2393|0.0753|0.8790|28.9335|

在colab上表现良好，可为什么在我本机却过拟合？

**Comment by zhaipro** [comment]
看来损失函数很重要呀

**Comment by zhaipro** [comment]
洗牌后的效果：
loss: 0.0016 - dice_coeff: 0.7766 - psnr: 35.3927 - val_loss: 6.2160e-04 - val_dice_coeff: 0.6873 - val_psnr: 35.8081

**Comment by zhaipro** [comment]
- 嘴唇：[80, 50, 230]
- 脸：[190, 90, 90]
- 舌头：[0, 255, 255]

**Comment by zhaipro** [comment]
惨痛的经验：开发环境还是交给conda吧，我不行啦，对不起pip

**Comment by zhaipro** [comment]
<div>
<img src="https://user-images.githubusercontent.com/8620842/82720814-a9e6a700-9ce9-11ea-9f6d-f59e1c15f28c.jpg" height="256" >
<img src="https://user-images.githubusercontent.com/8620842/82720817-ad7a2e00-9ce9-11ea-93c3-c8d0b1427487.jpg" height="256" >
<div>
数据增强已经足够了，只要不会过拟合就行，现在的目标是正确率啦。

**Comment by zhaipro** [comment]
代办项

1. 损失函数
1. 批量大小
1. 模型深度
1. 模型广度
1. Dropout

**Comment by zhaipro** [comment]
数据强化

- [x] 水平翻转
- [x] 缩放
- [x] 挪动
- [x] 模糊
- [x] 色彩抖动
- [ ] 侧脸
- [ ] 遮挡，蒙眼
- [ ] 老人
- [ ] 大眼睛？
- [ ] 根据复杂度打分

只要数据强，其他的都是小事儿。

**Comment by zhaipro** [comment]
Dropout能否帮助模型寻找最优解？

**Comment by zhaipro** [comment]
代办项

- [x] 评测XX的分割力度

没办法，我们的分割力度能到90%就不错啦，不求100分，但求最高分。

**Comment by zhaipro** [comment]
根据人脸扭转的程度来表示训练权重？

**Comment by zhaipro** [comment]
最令我开心的事儿就是发现自己是个脑残…

**Comment by zhaipro** [comment]
起始层不要提取太多的信息，不然容易过拟合。

**Comment by zhaipro** [comment]
![3970](https://user-images.githubusercontent.com/8620842/83724548-0c895c80-a673-11ea-9910-058f4e83d406.jpg)

**Comment by zhaipro** [comment]
hair cloth lip
96164 / 85977983 = 0.0011319825261650242

**Comment by zhaipro** [comment]
![720](https://user-images.githubusercontent.com/8620842/83840769-fe4e4580-a731-11ea-8d86-f542eaa07431.jpg)
